Automattic/wp-calypso
# Launchpad: Add task to unlock styles

Fixes https://github.com/Automattic/dotcom-forge/issues/1057

### Proposed Changes

Add a new step to the launchpad for users who have customized Global Styles during the newsletter flow. When clicking in the step, we show a modal so users can unlock the styles.

New step | Modal
--- | ---
<img width="250" alt="Screen Shot 2022-10-26 at 17 09 15" src="https://user-images.githubusercontent.com/1233880/198065204-e3f7de6e-47eb-4558-98a5-48eb692e2e22.png"> | <img width="500" alt="Screen Shot 2022-10-27 at 17 01 37" src="https://user-images.githubusercontent.com/1233880/198326103-565a94a6-d5d8-467d-9f1d-ac49bf9797cd.png">


⚠️ Note that the default color currently offered during the newsletter flow (Lettre) is not the default color provided by the theme, so the site will fallback to the default color unless the user upgrades to a paid plan. ⚠️

---

This PR also slightly simplify the `usePremiumGlobalStyles` hook to use a given site ID rather than trying to determine it. This way we avoid some API calls that makes the flow a bit smoother.

### Testing Instructions

- Use the Calypso live link below
- Go to `/setup/intro?flow=newsletter`
- Create a new site and pick the **Free** plan
- Once the site is created (after you reach the "Let’s add your first subscribers" screen), go to the blog RC and add the `wpcom-limit-global-styles` blog sticker
- Reload the browser page so the setup flow can fetch the new block sticker
- Proceed with the setup and stop once you reach the "Your Newsletter is ready!" screen
- Make sure there is a new "Activate advanced design customization tools" step
<img width="150" alt="Screen Shot 2022-10-26 at 17 09 15" src="https://user-images.githubusercontent.com/1233880/198070523-992fb21d-2a4e-48ee-baec-6030e709160e.png">

- Click on it
- Make sure this modal shows up:
<img width="400" alt="Screen Shot 2022-10-27 at 17 01 37" src="https://user-images.githubusercontent.com/1233880/198326103-565a94a6-d5d8-467d-9f1d-ac49bf9797cd.png">


- Purchase a paid plan and go back to the "Your Newsletter is ready!" screen (`/setup/launchpad?flow=newsletter&siteSlug=<slug>`)
- Make sure the "Activate advanced design customization tools" step has been completed
<img width="150" alt="Screen Shot 2022-10-26 at 17 07 50" src="https://user-images.githubusercontent.com/1233880/198070591-cc6e3cf6-6a4c-484f-a752-3bd21229e506.png">

- Disable the `limit-global-styles` feature flag by adding a `flags=-limit-global-styles` param to the URL
- Make sure the "Activate advanced design customization tools" step doesn't appear
<img width="150" alt="Screen Shot 2022-10-26 at 17 09 32" src="https://user-images.githubusercontent.com/1233880/198070652-03907042-b1ee-4f00-b3b4-3c4394dff2e7.png">
